src/grpc/dp_grpc_conv.cpp

```diff
@@ -203,19 +231,34 @@ void DpToGrpcFwrule(const struct dp_fwall_rule *dp_rule, FirewallRule *grpc_rule
 		grpc_rule->set_action(FirewallAction::DROP);
 
 	src_ip = new IpAddress();
-	src_ip->set_ipver(IpVersion::IPV4);
-	src_ip->set_address(GrpcConv::Ipv4ToStr(dp_rule->src_ip));
 	src_pfx = new Prefix();
+	if (dp_rule->src_ip.ip_type == RTE_ETHER_TYPE_IPV4) {
+		src_ip->set_ipver(IpVersion::IPV4);
+		src_ip->set_address(GrpcConv::Ipv4ToStr(dp_rule->src_ip.ipv4));
+		src_pfx->set_length(__builtin_popcount(dp_rule->src_mask.ip4));
+	} else {
+		src_ip->set_ipver(IpVersion::IPV6);
+		inet_ntop(AF_INET6, dp_rule->src_ip.ipv6, strbuf, sizeof(strbuf));
+		src_ip->set_address(strbuf);
+		src_pfx->set_length(DpIpv6MaskToPrefixLen(dp_rule->src_mask.ip6));
+	}
 	src_pfx->set_allocated_ip(src_ip);
-	src_pfx->set_length(__builtin_popcount(dp_rule->src_ip_mask));
 	grpc_rule->set_allocated_source_prefix(src_pfx);
 
 	dst_ip = new IpAddress();
-	dst_ip->set_ipver(IpVersion::IPV4);
-	dst_ip->set_address(GrpcConv::Ipv4ToStr(dp_rule->dest_ip));
 	dst_pfx = new Prefix();
+	if (dp_rule->dest_ip.ip_type == RTE_ETHER_TYPE_IPV4) {
+		dst_ip->set_ipver(IpVersion::IPV4);
+		dst_ip->set_address(GrpcConv::Ipv4ToStr(dp_rule->dest_ip.ipv4));
+		dst_pfx->set_length(__builtin_popcount(dp_rule->dest_mask.ip4));
+	} else {
+		dst_ip->set_ipver(IpVersion::IPV6);
+		inet_ntop(AF_INET6, dp_rule->dest_ip.ipv6, strbuf, sizeof(strbuf));
+		dst_ip->set_address(strbuf);
+		dst_pfx->set_length(DpIpv6MaskToPrefixLen(dp_rule->dest_mask.ip6));
+	}
+
```

Comment:
This is the third(?) time I see this block, I think it could be separated as a `GrpcConv::DpToGrpcPrefix` or something

Reply:
Done.